### DISCUSSION TITLE
How does this package actually work I am very confused?

### DISCUSSION THREAD

**Discussion by blnayan** [PrismarineJS/mineflayer#1635] [Q&A]
I have no idea how do these types of packages work so can someone smart enough, explain how these things work. Is there some kind of Minecraft API that mineflayer uses or am I completely wrong?

**Comment by ghost** [comment]
> I have no idea how do these types of packages work so can someone smart enough, explain how these things work. Is there some kind of Minecraft API that mineflayer uses or am I completely wrong?

Minecraft Protocol:

https://wiki.vg/Protocol

Mineflayer uses node-minecraft-protocol, it is a handy library for handling Minecraft packages on node js

https://github.com/PrismarineJS/node-minecraft-protocol

Packages are sent via TCP

**Comment by rom1504** [comment]
Mineflayer communicates with minecraft servers with the network, the same
as what the vanilla minecraft client is doing.
So for example it sends packets (a bunch of bytes: numbers between 0 and
255) saying "move there" or "dig that block" and the server tell it "that
block  changed" or "this entity moved".
